Products/CMFPlone/browser/recyclebin.py

```diff
@@ -669,6 +780,57 @@ def format_size(self, size_bytes):
         """Format size in bytes to human-readable format"""
         return human_readable_size(size_bytes)
 
+    def _get_redirect_url_after_restoration(self, restored_obj, target_container=None):
+        """Determine the appropriate URL to redirect to after item restoration.
+
+        This method implements a fallback strategy:
+        1. Try restored object's view URL first
+        2. If that fails, try target container URL
+        3. If that fails, use the recycle bin URL
```

Comment:
I don't understand the strategy. What is the situation where an item was restored but we can't redirect to its view URL?

Reply:
Umm, the implementation was thought inna way:-
- try to get the `{absolute_url()}/view` but if failing (maybe it is not available for the object)
- then get the target container url but if the worst case scenario - object is broken
- then target recyclebin url...

Reply:
> but if failing (maybe it is not available for the object)

This is the part I don't understand. Getting the object's absolute_url should always work. You can use restored_obj.restrictedTraverse("@@plone_context_state").view_url() to get it with the `/view` suffix for the content types that require that. (This comes from https://github.com/plone/plone.app.layout/blob/master/plone/app/layout/globals/context.py#L64)

Reply:
Maybe, I overcomplicated it too much... However, I have implemented a simplified based on ur suggestions in https://github.com/plone/Products.CMFPlone/pull/4175/commits/56428b176408451e30b9038268012c031311d48b